### PR TITLE
Grafana middleware

### PR DESCRIPTION
**Pull request by heyhalcyon**
# Grafana-Influxdb
A middleware between Grafana and Influxdb that has the following functions:  
- simple forwarding
- optimize queries, accelerate the frontend rendering speed.
- intelligently decide the sample size

A detailed description and report can be found in #650

### CODE REVIEW

**Comment by chenlica** [comment]
@heyhalcyon please add a detailed description or add a URL to the github issue with the description.  Also make sure NOT to include big data files in the codebase.  Just include some sample records.

File: sandbox/yitan-jiahao-grafana-middleware-2019/SampleRatio.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+import math
+
+from tslearn.piecewise import SymbolicAggregateApproximation
+from tslearn.utils import to_time_series
+
+from tslearn.preprocessing import TimeSeriesScalerMeanVariance
+from tslearn.preprocessing import TimeSeriesResampler
+
+
+##dbname = "NOAA_water_database"
```
Comment (AC-Unicorn) [inline sandbox/yitan-jiahao-grafana-middleware-2019/SampleRatio.py]: If the commended code is unnecessary, please delete them

**Comment by AC-Unicorn** [comment]
Please provide some information about this PRin the description section, like what is the purpose of these python files, how to run those files, library used, python version. Otherwise, please also include Issue link in the description section.

File: sandbox/yitan-jiahao-grafana-middleware-2019/SampleRatio.py (hunk 1 above)
Comment (heyhalcyon) [reply sandbox/yitan-jiahao-grafana-middleware-2019/SampleRatio.py]: This file is highly dependent on the dataset. We used it on 3 different datasets. The urls need to be modified every time we change a db, so it's more convenient to leave it there.

**Comment by heyhalcyon** [comment]
> Where are contents about how to run your code?

Sorry I forgot that part. I've added that in the readme file.

**Pull request merged**